### PR TITLE
Support default queues and priority groups

### PR DESCRIPTION
**Pull request by kcudnik**

### CODE REVIEW

File: syncd/syncd.cpp [hunk 1]
```diff
@@ -1248,10 +1248,12 @@ bool isVeryFirstRun()
 
 int main(int argc, char **argv)
 {
-    swss::Logger::getInstance().setMinPrio(swss::Logger::SWSS_NOTICE);
+    swss::Logger::getInstance().setMinPrio(swss::Logger::SWSS_DEBUG);
```
Comment (lguohan) [inline syncd/syncd.cpp]: keep it as notice?

File: syncd/syncd.cpp [hunk 2]
```diff
@@ -1248,10 +1248,12 @@ bool isVeryFirstRun()
...
 
     SWSS_LOG_ENTER();
 
+    swss::Logger::getInstance().setMinPrio(swss::Logger::SWSS_NOTICE);
+
```
Comment (lguohan) [inline syncd/syncd.cpp]: why do we need to setMinPrio here?

File: syncd/syncd_reinit.cpp [hunk 3]
```diff
@@ -610,6 +617,174 @@ void helperCheckVlanId()
...
+
+std::vector<sai_object_id_t> saiGetPortQueues(sai_object_id_t portId)
+{
+    SWSS_LOG_ENTER();
+
+    uint32_t queueCount = saiGetPortNumberOfQueues(portId);
+
+    std::vector<sai_object_id_t> queueList;
+
+    queueList.resize(queueCount);
```
Comment (lguohan) [inline syncd/syncd_reinit.cpp]: if the queueCount is zero, we can just return?

File: syncd/syncd_reinit.cpp [hunk 4]
```diff
@@ -610,6 +617,174 @@ void helperCheckVlanId()
...
+
+std::vector<sai_object_id_t> saiGetPortPriorityGroups(sai_object_id_t portId)
+{
+    SWSS_LOG_ENTER();
+
+    uint32_t pgCount =  saiGetPortNumberOfPriorityGroups(portId);
+
+    std::vector<sai_object_id_t> pgList;
+
+    pgList.resize(pgCount);
```
Comment (lguohan) [inline syncd/syncd_reinit.cpp]: if pgCount is zero, can we just return from here?

File: syncd/syncd.cpp (hunk 2 above)
Comment (kcudnik) [inline syncd/syncd.cpp]: i use it becasue i want "main" entry to be logged as debug

File: syncd/syncd_reinit.cpp (hunk 3 above)
Comment (kcudnik) [inline syncd/syncd_reinit.cpp]: not significant optimalization

Comment (kcudnik) [inline syncd/syncd_reinit.cpp]: and it will never be zero

File: syncd/syncd_reinit.cpp (hunk 4 above)
Comment (kcudnik) [inline syncd/syncd_reinit.cpp]: and it will never be zero

File: syncd/syncd_hard_reinit.cpp [hunk 5]
```diff
@@ -323,7 +327,56 @@ sai_object_id_t processSingleVid(sai_object_id_t vid)
...
+        {
+            // this is one of default priority group id
+            // we don't need to create it, just set attributes if any
+
+            rid = pgRid;
+
+            createObject = false;
+
+            SWSS_LOG_DEBUG("default priority group will not be created, processed VID %llx to RID %llx", vid, rid);
+        }
```
Comment (qiluo-msft) [inline syncd/syncd_hard_reinit.cpp]: The 3 branches above are quite similar, is it possible to reuse most of the code?

Comment (kcudnik) [inline syncd/syncd_hard_reinit.cpp]: not yet, after there will be need to add something similar

File: syncd/syncd_hard_reinit.cpp [hunk 6]
```diff
@@ -140,9 +140,11 @@ void checkAllIds()
         g_vidToRidMap.erase(it);
     }
 
-    if (g_vidToRidMap.size() != 0)
+    size_t size = g_vidToRidMap.size();
+
+    if (size != 0)
     {
-        SWSS_LOG_ERROR("vid to rid map is not empty after translation");
+        SWSS_LOG_ERROR("vid to rid map is not empty (%lu) after translation", size);
```
Comment (kcudnik) [inline syncd/syncd_hard_reinit.cpp]: fixed

File: syncd/syncd.cpp (hunk 1 above)
Comment (kcudnik) [inline syncd/syncd.cpp]: it is notice after ENTER

File: syncd/syncd_hard_reinit.cpp (hunk 6 above)
Comment (qiluo-msft) [inline syncd/syncd_hard_reinit.cpp]: Closed.

File: syncd/syncd_reinit.cpp (hunk 3 above)
Comment (lguohan) [inline syncd/syncd_reinit.cpp]: yes, on barefoot p4 switch, it is zero.

Comment (lguohan) [inline syncd/syncd_reinit.cpp]: I think it's better to skip if the count is zero. I am more worried about hitting bugs in the sai.

Comment (kcudnik) [inline syncd/syncd_reinit.cpp]: what ? is zero ? then why there is inconsistency whit that ? as you said before how we can't have queues ?

File: syncd/syncd_reinit.cpp [hunk 7]
```diff
@@ -64,6 +65,7 @@ std::vector<sai_object_id_t> saiGetPortList()
     attr.value.objlist.count = portCount;
     attr.value.objlist.list = portList.data();
 
+    // we assube port list is always returned in the same order
```
Comment (stcheng) [inline syncd/syncd_reinit.cpp]: assume

Comment (kcudnik) [inline syncd/syncd_reinit.cpp]: yea, so far there are many assumptions :P

Comment (stcheng) [inline syncd/syncd_reinit.cpp]: :D

**Pull request merged**